File: insight/src/main/java/info/nightscout/androidaps/plugins/pump/insight/app_layer/remote_control/SetTBRMessage.java
package info.nightscout.androidaps.plugins.pump.insight.app_layer.remote_control;

import info.nightscout.androidaps.plugins.pump.insight.app_layer.AppLayerMessage;
import info.nightscout.androidaps.plugins.pump.insight.app_layer.Service;
import info.nightscout.androidaps.plugins.pump.insight.descriptors.MessagePriority;
import info.nightscout.androidaps.plugins.pump.insight.utils.ByteBuf;

public class SetTBRMessage extends AppLayerMessage {

    private int percentage;
    private int duration;

    public SetTBRMessage() {
        super(MessagePriority.NORMAL, false, true, Service.REMOTE_CONTROL);
    }

    @Override
    protected ByteBuf getData() {
        ByteBuf byteBuf = new ByteBuf(6);
        byteBuf.putUInt16LE(percentage);
        byteBuf.putUInt16LE(duration);
        byteBuf.putUInt16LE(31);
        return byteBuf;
    }

    public void setPercentage(int percentage) {
        this.percentage = percentage;
    }

    public void setDuration(int duration) {
        this.duration = duration;
    }
}

File: insight/src/main/java/info/nightscout/androidaps/plugins/pump/insight/app_layer/remote_control/SetTBRMessage.kt
package info.nightscout.androidaps.plugins.pump.insight.app_layer.remote_control

import info.nightscout.androidaps.plugins.pump.insight.app_layer.AppLayerMessage
import info.nightscout.androidaps.plugins.pump.insight.app_layer.Service
import info.nightscout.androidaps.plugins.pump.insight.descriptors.MessagePriority
import info.nightscout.androidaps.plugins.pump.insight.utils.ByteBuf

class SetTBRMessage : AppLayerMessage(MessagePriority.NORMAL, false, true, Service.REMOTE_CONTROL) {

    internal var percentage = 0
    internal var duration = 0

    override val data: ByteBuf
        get() {
            val byteBuf = ByteBuf(6).apply {
                putUInt16LE(percentage)
                putUInt16LE(duration)
                putUInt16LE(31)
            }
            return byteBuf
        }
}